internal/common/customplanmodifier/non_updatable.go

```diff
@@ -0,0 +1,50 @@
+package customplanmodifier
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	planmodifier "github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+)
+
+func NonUpdatableStringAttributePlanModifier(attribute string) planmodifier.String {
+	return &nonUpdatableStringAttributePlanModifier{
+		Attribute: attribute,
+	}
+}
```

Comment:
```suggestion
func NonUpdatableStringPlanModifier() planmodifier.String {
	return &nonUpdatableStringAttributePlanModifier{}
}
```

no need to have the attribute name now, also suggestion for keeping name more concise

Reply:
Not technically needed, but I kept it so that the error message is clear on which attribute can't be updated. No strong opinion here, so if you still think it should be removed I can remove it

Reply:
Ohh I see, could we use `req.Path` to get this information?

Reply:
`req.Path` does have the info. THanks